### PR TITLE
[BREAKING] Feat: tx pause

### PR DESCRIPTION
**Pull request by sander2**
Breaking because of new events and removal of StatusCode::Shutdown

This adds a fork of substrate's WIP tx-pause pallet. I added some integration tests to ensure functionality, including one that shows that it shouldn't be possible to brick to parachain, i.e. democracy can always be used to recover. 

I also removed StatusCode::Shutdown from security and its related tests, since with this pr it becomes redundant

### CODE REVIEW

File: parachain/runtime/interlay/src/lib.rs [hunk 1]
```diff
@@ -168,27 +168,22 @@ impl Contains<RuntimeCall> for BaseCallFilter {
         if matches!(
             call,
             RuntimeCall::System(_)
-                | RuntimeCall::Authorship(_)
-                | RuntimeCall::Session(_)
                 | RuntimeCall::Timestamp(_)
-                | RuntimeCall::ParachainSystem(_)
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/lib.rs:174]: Why remove these? I'm not sure what would happen if these were paused, have you tested that?

**Review by gregdhill — COMMENTED** [review]
One comment on the change to the filter, otherwise LGTM!

File: parachain/runtime/interlay/src/lib.rs (hunk 1 above)
Comment (sander2) [reply parachain/runtime/interlay/src/lib.rs:174]: Good catch, it's a copy-paste mistake, I will fix. I copied from standalone, which doesn't have these pallets. There is an integration test for all pallets that are in the standalone runtime, but obviously that can't test these there. We'll be able to test these properly once we get rid of standalone but for now I think it's fine - just don't pause these pallets :)

Comment (gregdhill) [reply parachain/runtime/interlay/src/lib.rs:174]: @sander2 on a related note I noticed the other runtimes don't catch `RuntimeCall::Session(_)` in the filter, up to you if we add that there as well

**Review by gregdhill — APPROVED** [review]

**Pull request merged**